ff/serving/metadata/metadata.go

```diff
@@ -775,6 +793,47 @@ func NewMetadataServer(logger *zap.SugaredLogger) (*MetadataServer, error) {
 	}, nil
 }
 
+func NewMetadataTypesenseServer(logger *zap.SugaredLogger, host string, port string, apiKey string) (*MetadataServer, error) {
```

Comment:
Instead of doing this, I recommend creating a struct called: MetadataServerParams with two fields, logger and *TypesenseParams (another struct that contains host, port and apiKey)

If TypesenseParams is not nil, you should do this. Otherwise you shouldn't.

There should only be one NewMetadataServer function and it just takes a MetadataServerParams